File: src/main/java/com/itmo/microservices/shop/user/impl/repository/UserRepository.java
package com.itmo.microservices.shop.user.impl.repository;

import com.itmo.microservices.shop.user.impl.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserRepository extends JpaRepository<User, UUID> {
    User findByUsername(String username);
    Boolean existsUserByUsername(String username);
    Optional<User> findById(UUID uuid);
}

File: src/main/kotlin/com/itmo/microservices/shop/demo/users/impl/repository/UserRepository.kt
package com.itmo.microservices.shop.demo.users.impl.repository

import org.springframework.data.jpa.repository.JpaRepository
import com.itmo.microservices.shop.demo.users.impl.entity.AppUser
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<AppUser, String>